920470: include chars from rfc 2046

nerrehmit opened the pull request:
RFC 2046 allows additional chars for the boundary.
\d removed as it is covered by \w in the regex. 
Removed unnecessary escapes.

This should fix https://github.com/SpiderLabs/owasp-modsecurity-crs/issues/1409 by adding the missing characters for the boundary.

nerrehmit committed:
920470: include chars from rfc 2046

RFC 2046 allows additional chars for the boundary.
\d removed as it is covered by \w in the regex. Removed unnecessary
escapes.

nerrehmit committed:
920470: include chars from rfc 2046

RFC 2046 allows additional chars for the boundary.
\d removed as it is covered by \w in the regex. Removed unnecessary
escapes.

dune73 commented:
Thank you for the welcome contribution and fix @nerrehmit.

Would you mind adding an additional unit test or two for the new regex?

dune73 commented:
Assigning @fgsch as he had the initial idea to extend the list in the discussion around #1406.

fgsch commented:
@nerrehmit Did you have a chance to check the failure we discussed over slack?

fgsch commented:
@nerrehmit Can you also rebase this against v3.3/dev?

nerrehmit committed:
Merge branch 'v3.2/dev' of github.com:nerrehmit/owasp-modsecurity-crs into v3.2/dev

nerrehmit committed:
Rearange characters and add positive and negative test cases.

Moved the dash to the end of the character set to avoid escaping it.
Added test with all the new characters and a test for multiple whitespaces.
Allowed a previously blocked charset.

fgsch commented:
Thanks for updating it. I will take a look asap.

dune73 commented:
Meeting outcome: @franbuehler is taking this over.

franbuehler reviewed (approved):
PR works as advertised. Tests are passing. I will merge this PR.

franbuehler commented:
Thank you @nerrehmit for your first contribution to our project. 
Sorry it took so long.
We look forward to your next contribution :-)
